### ISSUE TITLE
[BUG] Kwarg Mask constructor fails

### ISSUE THREAD

**Issue by georgematheos** [genjax-community/genjax#1490]
```python
from genjax import Mask
Mask(value=jnp.arange(10), flag=jnp.ones(10, dtype=bool))
```
throws an error.

If the interface for interacting with Mask objects is to use the `.value` and `.flag` fields, I think that it would make sense for this kwarg constructor to work.
